common/queue/src/main/java/org/thingsboard/server/queue/discovery/ZkDiscoveryService.java

```diff
@@ -281,11 +302,12 @@ public void childEvent(CuratorFramework curatorFramework, PathChildrenCacheEvent
             case CHILD_ADDED:
             case CHILD_UPDATED:
             case CHILD_REMOVED:
-                partitionService.recalculatePartitions(serviceInfoProvider.getServiceInfo(), getOtherServers());
+                TransportProtos.ServiceInfo currentService = serviceInfoProvider.getServiceInfo();
+                partitionService.recalculatePartitions(currentService, getOtherServers());
+                eventPublisher.publishEvent(new ServiceListChangedEvent(getAllServers(), currentService, instance));
```

Comment:
Please publish this event in the HashPartitionService.recalculatePartitions. It is synchronized.